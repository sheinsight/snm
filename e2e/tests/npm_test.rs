use std::env::current_dir;

e2e::test1! {
  #[tokio::test]
  test_show_npm_version_when_missing_default_node_and_npm,
  cwd: current_dir()?.join("tests").join("fixtures").join("empty"),
  envs: [],
  |builder:e2e::CommandBuilder| => {
    builder.add_snapshot("npm -v")?;
    builder.assert_snapshots(|name,res| {
      insta::assert_snapshot!(name, res);
    })?;
  }
}

e2e::test1! {
  #[tokio::test]
  test_show_npm_version_when_default_npm_missing_but_node_exists,
  cwd: current_dir()?.join("tests").join("fixtures").join("empty"),
  envs: [],
  |builder:e2e::CommandBuilder| => {
    builder.add_snapshot("snm node install 20.0.0")?;
    builder.add_snapshot("snm node default 20.0.0")?;
    builder.add_snapshot("npm -v")?;
    builder.assert_snapshots(|name,res| {
      println!("name---->: {:?}", name);
      println!("res---->: {:?}", res);
      insta::assert_snapshot!(name, res);
    })?;
  }
}
