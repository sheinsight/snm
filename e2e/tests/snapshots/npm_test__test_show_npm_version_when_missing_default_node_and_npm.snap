---
source: e2e/tests/npm_test.rs
expression: res
---
name: "test_show_npm_version_when_missing_default_node_and_npm"

id: 1
is: npm -v

Error: No default Node.js version found
